### ISSUE TITLE
Expected browser binary location, but unable to find binary in default location

### ISSUE THREAD

**Issue by adigoldin22** [InstaPy/InstaPy#6070]
i have tried all possible solutions but i haven't been able to resolve this issue 
please help guys

InstaPy Version: 0.6.13
 ._.  ._.  ._.  ._.  ._.  ._.  ._.  ._. 
Workspace in use: "C:/Users/ADI/InstaPy"
Traceback (most recent call last):
  File "C:\Users\ADI\Pictures\instapy-quickstart-master\instapy-quickstart-master\quickstart_templates\basic_follow-unfollow_activity.py", line 30, in <module>
    session = InstaPy(username=insta_username,
  File "C:\Users\ADI\AppData\Local\Programs\Python\Python38-32\lib\site-packages\instapy\instapy.py", line 325, in __init__
    self.browser, err_msg = set_selenium_local_session(
  File "C:\Users\ADI\AppData\Local\Programs\Python\Python38-32\lib\site-packages\instapy\browser.py", line 123, in set_selenium_local_session
    browser = webdriver.Firefox(
  File "C:\Users\ADI\AppData\Local\Programs\Python\Python38-32\lib\site-packages\selenium\webdriver\firefox\webdriver.py", line 170, in __init__
    RemoteWebDriver.__init__(
  File "C:\Users\ADI\AppData\Local\Programs\Python\Python38-32\lib\site-packages\selenium\webdriver\remote\webdriver.py", line 157, in __init__
    self.start_session(capabilities, browser_profile)
  File "C:\Users\ADI\AppData\Local\Programs\Python\Python38-32\lib\site-packages\selenium\webdriver\remote\webdriver.py", line 252, in start_session
    response = self.execute(Command.NEW_SESSION, parameters)
  File "C:\Users\ADI\AppData\Local\Programs\Python\Python38-32\lib\site-packages\selenium\webdriver\remote\webdriver.py", line 321, in execute
    self.error_handler.check_response(response)
  File "C:\Users\ADI\AppData\Local\Programs\Python\Python38-32\lib\site-packages\selenium\webdriver\remote\errorhandler.py", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.SessionNotCreatedException: Message: Expected browser binary location, but unable to find binary in default location, no 'moz:firefoxOptions.binary' capability provided, and no binary flag set on the command line

**Comment by Xsnipe2Z** [comment]
I have The same issue please someone help

**Comment by Jasiri-w** [comment]
### My Solution

`session = InstaPy(username="username", password="password", browser_executable_path=r"C:\Program Files\Mozilla Firefox\firefox.exe").login()`

**Pass** _browser_executable_path = r"C:\[Custom Location]"_ in your Session instance.

Reinstalling Firefox did not work for me.

### Explanation

`selenium.common.exceptions.SessionNotCreatedException: Message: Expected browser binary location, but unable to find binary in default location, no 'moz:firefoxOptions.binary' capability provided, and no binary flag set on the command line`

That message implies that the GeckoDriver was unable to locate the Firefox binary _(firefox.exe)_ while trying to initiate/spawn a new browsing context i.e. Firefox browser-session.

**Why?**
Probably for 1 of 2 reasons:

1. Firefox isn't installed in your system.
2. Firefox isn't installed in the default location within your system.

Normally, any Selenium project would take...
 
`options = webdriver.firefox.options.Options()
options.binary_location = r"C:\Program Files\Mozilla Firefox\firefox.exe"
browser = webdriver.Firefox(options=options)`

for a customized location passing the absolute path of the firefox binary through an Options() instance, but since InstaPy handles initializing the browser, you must pass this through the class attribute "browser_executable_path".

( How the InstaPy InstaPy class initilalizes )
![image](https://user-images.githubusercontent.com/6060064/114033659-1cbb2b00-9886-11eb-85ef-ae209e0ead15.png)

TL;DR: Adding the parameter browser_executable_path of the InstaPy class allowed me to pass the r"C:\[Custom Firefox Location]" as the firefox location.

**References**

- https://stackoverflow.com/questions/64034859/invalidargumentexception-message-binary-is-not-a-firefox-executable-error-usin/64035303#64035303
- https://stackoverflow.com/questions/66000401/expected-browser-binary-location-but-unable-to-find-binary-in-default-location/66000618#66000618
- https://stackoverflow.com/questions/65318382/expected-browser-binary-location-but-unable-to-find-binary-in-default-location/65322308#65322308
